### DISCUSSION TITLE
Grabbing the duration of a Hyperdeck clip

### DISCUSSION THREAD

**Discussion by randallpacker** [danielbuechele/atemOSC#276] [Ideas]
I have a Hyperdeck request that I believe would be a real game changer. If it is possible to build into atemOSC a "disk list: slot id" request (see below), then provides the duration of a clip so you can anticipate its end before it happens in order to initiate a time-based transition. This would be huge! Whereas it is now possible in atemOSC to grab the state change from 'play' to 'idle', that happens at the very end of a clip when it is too late to trigger a smooth mix transition. Is this something that could be put on the docket?  I have included the information below from the Hyperdeck protocol documentation. Thanks!! Randall

_Retrieving clip information   The "disk list" command returns the information for each playable clip on a given disk. Without parameters, the command returns information for the current active disk: disk list↵ If a slot id is specified, the disk in that slot will be queried: disk list: slot id: {Slot ID}↵ 

The server responds with the list of all playable clips on the disk in the format of: Index, name, formats, and **duration** in timecode: 206 disk list:↵ slot id: {Slot ID}↵ {clip index}: {name} {file format} {video format} {Duration timecode}↵ {clip index}: {name} {file format} {video format} {Duration timecode}↵ … ↵_

**Comment by SteffeyDev** [comment]
Related: https://github.com/SteffeyDev/atemOSC/issues/212
Still too busy to work on this project right now, but hoping to get back to it eventually, maybe in a few months

**Comment by randallpacker** [comment]
Thanks Peter for the update. Are you the only one working on the project?

**Reply by SteffeyDev** [reply]
Yes. Anyone is welcome to work on it, but I don't know of anyone else doing so right now.
